core/editor/src/dispatcher/events.rs

```diff
@@ -18,6 +18,34 @@ pub enum Response {
 
 #[derive(Debug, Clone)]
 pub struct Trace(Vec<MouseState>);
+
+impl Trace {
+	pub fn new() -> Self {
+		Self(vec![])
+	}
+	pub fn first_point(&self) -> Option<&MouseState> {
+		if !self.0.is_empty() {
+			Some(&self.0[0])
+		} else {
+			None
+		}
+	}
```

Comment:
You can just use `self.0.get(0)` instead

Reply:
even better: `Vec::first`